types/query_id.hh

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "utils/UUID.hh"
+
+using query_id = utils::tagged_uuid<struct query_id_tag>;
```

Comment:
`types/` directory is for storage system type system, not for random utility types we use. I think `query-request.hh` is a better place for this definition.

Reply:
ok, query-result.hh include query-request.hh so that could work.

Reply:
Done in v3